lib/tapioca/compilers/symbol_table/symbol_generator.rb

```diff
@@ -154,16 +159,22 @@ def compile_object(tree, name, value)
             name_of(klass)
           end
 
+          comments = documentation_comments(name)
+
           if klass_name == "T::Private::Types::TypeAlias"
-            tree << RBI::Const.new(name, "T.type_alias { #{T.unsafe(value).aliased_type} }")
+            constant = RBI::Const.new(name, "T.type_alias { #{T.unsafe(value).aliased_type} }")
+            constant.comments.concat(comments) unless comments.empty?
+            tree << constant
```

Comment:
You should be able to pass it directly to the constructor (and so avoid initializing an empty array just to concat to it):
```suggestion
            tree << RBI::Const.new(name, "T.type_alias { #{T.unsafe(value).aliased_type} }", comments: comments)
```

See the initializer hereL https://github.com/Shopify/rbi/blob/main/lib/rbi/model.rb#L303.